pynoc/cisco.py

```diff
@@ -79,68 +84,53 @@ def connect(self):
 
         :return:
         """
-        self._client = SSHClient()
-        self._client.set_missing_host_key_policy(AutoAddPolicy())
-        self._client.connect(self._host,
-                             username=self._username,
-                             password=self._password,
-                             allow_agent=False,
-                             look_for_keys=False)
-        self._shell = self._client.invoke_shell()
-        self._ready = True
-        output = self._send_command(
-            self.CMD_CARRIAGE_RETURN, self.CMD_LOGIN_SIGNALS
+        self._client = ConnectHandler(
+            device_type='cisco_ios', ip=self._host, username=self._username,
+            password=self._password
         )
-        if output.find('>') > 0:
-            self._enable_needed = True
-            self._ready = False
-        else:
-            self.set_terminal_length()
 
     def disconnect(self):
         """Disconnect from the switch.
 
         :return:
         """
         if self._client is not None:
-            self._client.close()
+            self._client.disconnect()
             self._client = None
-            self._shell = None
-            self._ready = False
 
     def enable(self, password):
         """Put the switch in enable mode.
 
         :param password: password to use to enable
         :return:
         """
-        if self.connected and self._enable_needed:
-            self._send_command(self.CMD_ENABLE, self.CMD_ENABLE_SIGNALS)
-            self._send_command(
-                password, self.CMD_GENERIC_SIGNALS, log=False
-            )
-            self._ready = True
-            self.set_terminal_length()
+        if self.connected:
+            self.disconnect()
+
+        self._client = ConnectHandler(
```

Comment:
The difference here is the `secret`.  However I'll change this to use the property instead.